### PR TITLE
allow default jax-rs packages to be overriden

### PR DESCRIPTION
**Pull request by johnmcclean-aol**
Allow default jax-rs packages to be overriden using configurable module.

e.g.
 
ConfigurableModule.builder().defaultJaxRsPackages(asList("new packages","go here").build();

also can choose between extending and overwriting the defaults

To overwrite everything : 

ConfigurableModule.builder().resetAll(true).defaultJaxRsPackages(asList("new packages","go here").build();

### CODE REVIEW

**Comment by edeegan** [comment]
Looks good

File: micro-core/src/main/java/com/aol/micro/server/utility/HashMapBuilder.java [hunk 1]
```diff
@@ -25,6 +28,15 @@ public Builder(K key, V value) {
 			build = new HashMap<K, V>();
 			build.put(key, value);
 		}
+		public Builder(Map<K,V> map) {
+			build = new HashMap<K, V>();
```
Comment (quike) [inline micro-core/src/main/java/com/aol/micro/server/utility/HashMapBuilder.java]: From official documentation: "An instance of HashMap has two parameters that affect its performance: initial capacity and load factor. [...] When the number of entries in the hash table exceeds the product of the load factor and the current capacity, the hash table is rehashed [...] If many mappings are to be stored in a HashMap instance, creating it with a sufficiently large capacity will allow the mappings to be stored more efficiently than letting it perform automatic rehashing as needed to grow the table."

If we are building here a copy of the passed map, I would suggest to initialize it with the proper size.

File: micro-core/src/test/java/app/async/com/aol/micro/server/SimpleApp.java [hunk 2]
```diff
@@ -20,4 +27,120 @@ public String myEndPoint(){
 		return "hello world!";
 	}
 	
+	public void test2(){
```
Comment (quike) [inline micro-core/src/test/java/app/async/com/aol/micro/server/SimpleApp.java]: Better test naming would be appreciated, so does to follow code style and avoid this indentation or big empty gaps/lines.

Comment (johnmcclean-aol) [inline micro-core/src/test/java/app/async/com/aol/micro/server/SimpleApp.java]: Good spot - those should not be there at all!

File: micro-core/src/main/java/com/aol/micro/server/utility/HashMapBuilder.java (hunk 1 above)
Comment (johnmcclean-aol) [inline micro-core/src/main/java/com/aol/micro/server/utility/HashMapBuilder.java]: Yeah - I think 

      build = new HashMap<K, V>(map);

is better in that regard and cleaner.

It's probably not a copy, as the builder is initialized with the map, but calling code can add more values. We could add an immutableBuild method that did this though (and returned an immutable map or view on the map).

At the moment, it's a convenience class to make configuration simpler, and isn't used in anything performance critical.

**Comment by johnmcclean-aol** [comment]
Merging in pull request, will update any further change requests on master.

**Pull request merged**